packages/evm/src/verkleAccessWitness.ts

```diff
@@ -377,3 +389,57 @@ export class VerkleAccessWitness implements VerkleAccessWitnessInterface {
     }
   }
 }
+
+export const generateExecutionWitness = async (
+  stateManager: StatefulVerkleStateManager | StatelessVerkleStateManager,
+  accessWitness: VerkleAccessWitness,
+  parentStateRoot: Uint8Array,
+) => {
+  const trie = (stateManager as StatefulVerkleStateManager)['_trie'] as VerkleTree
+  const postStateRoot = await stateManager.getStateRoot()
+  const ew: VerkleExecutionWitness = {
+    stateDiff: [],
+    parentStateRoot: bytesToHex(parentStateRoot),
+    verkleProof: undefined as any,
+  }
+  const accessedSuffixes = new Map<PrefixedHexString, number[]>()
+  for (const chunkKey of accessWitness['chunks'].keys()) {
+    const stem = chunkKey.slice(0, 64) as PrefixedHexString
+    if (accessedSuffixes.has(stem)) {
+      const suffixes = accessedSuffixes.get(stem)
+      suffixes!.push(parseInt(chunkKey.slice(64), 16))
+      accessedSuffixes.set(stem, suffixes!)
+    } else {
+      accessedSuffixes.set(stem, [parseInt(chunkKey.slice(64), 16)])
+    }
+  }
+
+  // Get values
+  for (const stem of accessedSuffixes.keys()) {
+    trie.root(parentStateRoot)
+    const suffixes = accessedSuffixes.get(stem)
+    if (suffixes === undefined || suffixes.length === 0) continue
+    const currentValues = await trie.get(hexToBytes(stem), accessedSuffixes.get(stem)!)
+    trie.root(postStateRoot)
+    const newValues = await trie.get(hexToBytes(stem), accessedSuffixes.get(stem)!)
+    const stemStateDiff = []
+    for (let x = 0; x < suffixes.length; x++) {
+      // skip if both are the same
+      if (
+        notNullish(currentValues[x]) &&
+        notNullish(newValues[x]) &&
+        equalsBytes(currentValues[x]!, newValues[x]!)
+      )
+        continue
+      stemStateDiff.push({
+        suffix: suffixes[x],
+        currentValue: currentValues[x] ? bytesToHex(currentValues[x]!) : null,
+        newValue: newValues[x] ? bytesToHex(newValues[x]!) : null,
+      })
+    }
```

Comment:
I think we could remove the non-null assertions (!) by just assigning these values to variables: for e.g.
```suggestion
      // skip if both are the same
      const currentValue = currentValues[x]
      const newValue = newValues[x]
      if (
        notNullish(currentValue) &&
        notNullish(newValue) &&
        equalsBytes(currentValue, newValue)
      )
        continue
      stemStateDiff.push({
        suffix: suffixes[x],
        currentValue,
        newValue,
      })
    }
```

Reply:
Could also maybe do without the notNullish helper by just checking `instanceof Uint8Array` for both?